### PR TITLE
Hc 121

### LINKED ISSUE #121: TypeError during 'consistency of volumedriver' check
### Problem description
While running the attended healthcheck, I received the following output:

```
[INFO] Checking consistency of volumedriver vs. ovsdb for vPool 'myvpool01': 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/celery/local.py", line 188, in __call__
    return self._get_current_object()(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "ovs/lib/healthcheck.py", line 64, in check_attended
    return HealthCheckController.execute_check(unattended, silent_mode)
  File "/usr/lib/python2.7/dist-packages/celery/local.py", line 188, in __call__
    return self._get_current_object()(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "ovs/lib/healthcheck.py", line 99, in execute_check
    HealthCheckController.check_openvstorage(logger)
  File "/usr/lib/python2.7/dist-packages/celery/local.py", line 188, in __call__
    return self._get_current_object()(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "ovs/lib/healthcheck.py", line 131, in check_openvstorage
    OpenvStorageHealthCheck.check_model_consistency(logger)
  File "ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py", line 732, in check_model_consistency
    "'{0}': {1}".format(vp.name, ', '.join(missinginvolumedriver)),
TypeError: sequence item 0: expected string, NoneType found
```

### Possible root of the problem
The error points out to the ''.join(list). When a list entry is None this can occur.
In my case I added a vdisk on a vpool that had a faulty preset. This vdisk is unregistered within the volumedriver due to that. OVS DAL has an entry 'None' in the volume_id of a vdisk.

### Possible solution
Catch this error and output it as a vdisk that has been created on a vpool with faulty preset.

### PR DESCRIPTION
**Pull request by JeffreyDevloo**
Fixes #121

### CODE REVIEW

File: ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py [hunk 1]
```diff
@@ -714,31 +714,32 @@ def check_model_consistency(logger):
...
+                        missing_in_model.append(voldrv_id)
 
                 # display discrepancies for vPool
-                if len(missinginvolumedriver) != 0:
-                    logger.warning("Detected volumes that are MISSING in volumedriver but ARE in ovsdb in vPool "
-                                   "'{0}': {1}".format(vp.name, ', '.join(missinginvolumedriver)),
-                                   'discrepancies_ovsdb_{0}'.format(vp.name))
+                if len(missing_in_volumedriver) != 0:
+                    logger.warning("Detected volumes that are MISSING in volumedriver but ARE in ovsdb in vPool (a known cause is a faulty preset) "
+                                   "'{0}': {1} ".format(vp.name, ', '.join([vdisk.guid for vdisk in missing_in_volumedriver])),
```
Comment (kinvaris) [inline ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py]: vdisk guid should be appended during crossreference, not during display

Comment (kinvaris) [inline ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py]: Also add that this is the model GUID and not a volume GUID

File: ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py [hunk 2]
```diff
@@ -714,31 +714,32 @@ def check_model_consistency(logger):
...
+                                    'discrepancies_ovsdb_{0}'.format(vp.name))
+
                 else:
                     logger.success("NO discrepancies found for ovsdb in vPool '{0}'".format(vp.name),
                                    'discrepancies_ovsdb_{0}'.format(vp.name))
 
-                if len(missinginmodel) != 0:
+                if len(missing_in_model) != 0:
```
Comment (kinvaris) [inline ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py]: Add that this is a volumedriver GUID, not a model guid.

**Comment by JeffreyDevloo** [comment]
Possible output now
```
[WARNING] Detected volumes that are MISSING in volumedriver but ARE in ovsdb in vPool (a known cause is a faulty preset) vpool name: myvpool01 - vdisk guid(s):4ef3626e-2138-4a7a-82c7-d261d1b4af5b
```

**Pull request merged**